lib/database/game.php

```diff
@@ -936,19 +937,31 @@ function getGameTopAchievers($gameID, $offset, $count, $requestedBy, $type = 0)
                   AND ach.Flags = 3 
                   AND gd.ID = $gameID
                 GROUP BY aw.User
-                $havingQuery
-                ORDER BY TotalScore DESC, LastAward $order
-                LIMIT $offset, $count";
+                ORDER BY TotalScore DESC, LastAward ASC";
 
     $dbResult = s_mysql_query($query);
     SQL_ASSERT($dbResult);
 
     if ($dbResult !== false) {
         while ($data = mysqli_fetch_assoc($dbResult)) {
-            $retval[] = $data;
+            if (count($high_scores) < 10) {
+                array_push($high_scores, $data);
+            }
+
+            if ($data['TotalScore'] == $mastery_score) {
+                if (count($masters) == 10) {
+                    array_shift($masters);
+                }
+                array_push($masters, $data);
+            } elseif (count($high_scores) == 10) {
+                break;
```

Comment:
keep the last ten rows matching the mastery score (they'll be the latest masters)